File: inf2100/ukesoppgaver/uke01/oppgave3.java
class Fib {
	final static int n = 40;

	static int fib1(int x) {
		int f1 = 0, f2 = 1;

		for (int i = 0; i < x; i++) {
			int f3 = f1 + f2;
			f1 = f2; 
			f2 = f3;
		}
		return f1;
	}

	static int fib2(int x) {
		if (x <= 2)
			return 1;
		else
			return fib2(x-2) + fib2(x-1);
	}

	public static void main(String arg[]) {
		System.out.println("fib1(" + n + ") = " + fib1(n));
		System.out.println("fib2(" + n + ") = " + fib2(n));
	}
}

File: inf2100/ukesoppgaver/oppgave3.pas
program Fib;

const N = 40;

function Fib1(x : Integer): Integer;
var 
	f1 : Integer; 
	f2 : Integer;
	f3 : Integer;
	i : Integer
begin
	f1 := 0;
	f2 := 1;
	i := 0;
	while  i < x do 
	begin
		f3 := f1 + f2;
		f1 := f2;
		f2 := f3;
	end;
	Fib1 := f1;
end; {Fib1}

function Fib2(x : Integer): Integer;
begin
	if x <= 2 then
		Fib2 := 1
	else
		Fib2 := Fib2(x-2) + Fib2(x-1)
end;

begin
	write(Fib1(N), eol)
	write(Fib2(N), eol)
end;
